src/recordlinker/models.py

```diff
@@ -0,0 +1,41 @@
+import uuid
+
+from sqlalchemy import orm, ForeignKey, String, JSON
+
+
+class Base(orm.DeclarativeBase):
+    pass
+
+
+class Person(Base):
+    __tablename__ = "mpi_person"
+
+    id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
+    internal_id: orm.Mapped[uuid.UUID] = orm.mapped_column(default=uuid.uuid4)
+
+
+class ExternalPerson(Base):
+    __tablename__ = "mpi_external_person"
+    # Create a unique index on the combination of person_id and external_id
+    __table_args__ = (orm.UniqueConstraint("person_id", "external_id"))
+
+    id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
+    person_id: orm.Mapped[int] = orm.mapped_column(ForeignKey("mpi_person.id"))
+    external_id: orm.Mapped[str] = orm.mapped_column(String(255))
+
+
+class Patient(Base):
+    __tablename__ = "mpi_patient"
+
+    id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
+    person_id: orm.Mapped[int] = orm.mapped_column(ForeignKey("mpi_person.id"))
+    data: orm.Mapped[dict] = orm.mapped_column(JSON)
+
+
+class BlockingKey(Base):
+    __tablename__ = "mpi_blocking_key"
+
+    id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
+    patient_id: orm.Mapped[int] = orm.mapped_column(ForeignKey("mpi_patient.id"))
+    key: orm.Mapped[str] = orm.mapped_column(String(50), index=True)
```

Comment:
Why `String(50)` rather than `String(255)` above?

Reply:
This table will get rather large having the potential for 7x rows for each Patient.  I think we should limit the size of these columns to what we actually need and nothing more.

Since we've started the conversation about limiting the number of blocking keys to a set number (maybe 10 or so), it might make more sense to break out a lookup table here.  A table to store the 10 or so blocking keys that we allow, then have another table like BlockingValue that stores the actually indexes for Patients.

Reply:
I think that makes sense. To make sure I'm understanding, what would be the advantages of creating a separate BlockingValue lookup table (e.g., performance-wise, readability, etc.)?